proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,208 @@
+
+#### Variant #1: `component` prims are still the only leaves
+We can say that it's okay that the "EastRiver" is now considered a `group`. All we've said is that it "might" contain models. The vast majority of prims in the scene graph will still be skipped during model hierarchy searches.
+
+The biggest drawback is when you considered material networks introduced at the assembly level, there could be a large number of prims to traverse. It may not make sense to nudge these materials into `component` models to prune traversal.
+
+#### Variant #2: Introduce a new `auto` kind which enables propagation
+`auto` becomes `kind`'s new fallback value. The "EastRiver" could be explicitly tagged with `kind=""` to preserve the current behavior. Propagation for `group` or `assembly` kinds is `group`. All other kinds propagate `""`.
+
+#### Variant #3: Use authored state to control propagation
+Only propagate when `kind` is unauthored. This is effectively the same as `auto` in that "EastRiver" can be tagged with `kind=""` to suppress propagation.
+
+In user interfaces, it may be hard to disambiguate between unauthored (and propagating) and authored (propagating) `kind` state. There would also not be way to author a value to re-enable propagation, but this may be fine.
+#### Recommendation
+This proposal advocates #2 or #3 as being the best choice as they minimize impact to current usage. The only thing that would have to change is `kind=""` would have to be explicitly authored to explicitly stop propagation. The biggest risk to either of these usages is that code sites may be doing explicit `GetMetadata("kind")` calls and comparisons instead of using the `UsdModelAPI`. For clarity, some of these variations may benefit from the concept to author a more explicit model hierarchy block-- say `kind="none"`, but no proposal strictly requires introducing this.
+
+### Instancing
+Propagation when `instanceable=True` is complicated, as a prim may have multiple parents with different model hierarchy validity. There are actually [bugs](https://github.com/PixarAnimationStudios/USD/issues/2406) with this today.
+
+The USD development team proposes that prim flags are considered in concert with composition arcs when constructing prototypes to deal with this ambiguity.
+
+Until that fix is available, the best that can be done in both the current state of the world and with self propagating hierarchies is to encourage users to be intentional about model hierarchy when authoring `instanceable` to avoid violating the hierarchy's continuity.
+
+### Additional API
+
+To further simplify usage of model hierarchy for developers, this proposal also advocates introducing `IsAssembly`, `IsComponent`, and `IsSubcomponent` methods on `UsdModelAPI`, along with associated cached prim flags and traversal predicates.
+
+This proposal also recommends `IsModel` should be deprecated in favor of `IsModelHierarchy` and should be equivalent to (`IsGroup() || IsComponent()`).
```

Comment:
I think that would be an improvement, and prevent the UB we get today from `kind = “model”` . Unfortunately, it’ll be tricky for Pixar to make this change given the historical uses of kind in our pipeline… I think that implementing the [“mix-in kinds” proposed on USD-interest](https://groups.google.com/g/usd-interest/c/S8pOpqr52DI/m/yFNNghUBBwAJ) would help us make the transition away from model and derived-from-model kinds in our assets.

Reply:
Would the resolution here be to remove deprecation recommendation?

Reply:
Renaming `IsModel` to `IsModelHierarchy` doesn't necessarily mean that `kind=model` has to go away. I can update with that note.